Makefile

```diff
@@ -56,5 +56,15 @@ test: test-unit ## Run the tests
 test-unit: ## Run the unit tests
 	$(Q)go test -count=1 -short ./...
 
+deploy: generate ## Deploy the Combo operator to the current cluster
+	kubectl apply -f manifests/crds			  && \
+	kubectl apply -f manifests/namespace.yaml && \
+	kubectl apply -f manifests/deploy
```

Comment:
There is no need to call `kubectl apply` multiple times. If you want to apply all files in the `/manifests` dir you could use:
```
$  k apply --recursive -f manifests
customresourcedefinition.apiextensions.k8s.io/combinations.combo.io created
customresourcedefinition.apiextensions.k8s.io/templates.combo.io created
clusterrole.rbac.authorization.k8s.io/combo-operator created
clusterrolebinding.rbac.authorization.k8s.io/combo-operator created
deployment.apps/combo-operator created
serviceaccount/combo-operator created
namespace/combo unchanged
```

Alternatively, if you want to apply the content at the existing paths, you can use this command:
```
k create -f manifests/namespace.yaml -f manifests/crds -f manifests/deploy
namespace/combo created
customresourcedefinition.apiextensions.k8s.io/combinations.combo.io created
customresourcedefinition.apiextensions.k8s.io/templates.combo.io created
clusterrole.rbac.authorization.k8s.io/combo-operator created
clusterrolebinding.rbac.authorization.k8s.io/combo-operator created
deployment.apps/combo-operator created
serviceaccount/combo-operator created
```

Reply:
This advice applies here as well: https://github.com/operator-framework/combo/pull/24/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R64-R67

Reply:
The main reason I avoided doing this is because the order of these being created matters:

```
customresourcedefinition.apiextensions.k8s.io/combinations.combo.io created
customresourcedefinition.apiextensions.k8s.io/templates.combo.io created
clusterrole.rbac.authorization.k8s.io/combo-operator created
clusterrolebinding.rbac.authorization.k8s.io/combo-operator created
namespace/combo created
Error from server (NotFound): error when creating "manifests/deploy/deployment.yaml": namespaces "combo" not found
Error from server (NotFound): error when creating "manifests/deploy/service_account.yaml": namespaces "combo" not found
```
Is there a way to get around this problem with this recursive solution?

Reply:
You can get around that error by ensuring the namespace YAML manifest is lexicographically lesser than the manifests that require it (basically any of the namespaced-scoped controller resources).

Reply:
More concretely, using prefixes:

- 01_namespace.yaml
- 02_deployment.yaml
- 02_serviceaccount.yaml
...

Reply:
Another alternative might be to remove the namespace parameter entirely, and let kubelet apply the resources in whatever the current (or default) context has specified for a namespace. The only caveat would be RBAC that wants an explicit namespace parameter for namespace-scoped tenants, but I don't know off-hand whether that's a required field.

Reply:
The namespace is required for the RBAC so I believe we'd have to go with the lexicographical ordering here.

Reply:
I would suggest one of the following solutions:
1. Change the `/manifests/namespace.yaml` to `/manifests/00_namespace.yaml` so it is applied before either of the dirs that you created.
2. Put all the yaml in `/manifests` and add prefixes to control ordering of application.

Personally, I lean towards option 2.

Reply:
I think I largely lean towards option 2 as well but I'd like to propose another alternative: we could just settle on a Makefile target that simply runs a bash wrapper script, utilizing envsubst to provide an absolutely basic templating layer without introducing additional tooling (helm, kustomize, etc.) while we iterate further. That way, we can at least have control over the $NAMESPACE that gets created by overriding the default value locally. Any thoughts?

Reply:
Seems reasonable, but can that be reported as an issue and addressed at a later date?

Reply:
Sorry - I had only glimpsed at your comment. You are effectively proposing that the namespace should be configurable via a bash script, and that the bash script should be in charge of ordering the order that resources are applied. That seems useful to get in earlier rather than later.

Reply:
Hmm - could we use combo to substitute the $NAMESPACE variable in the manifests being generated?

Reply:
Let's just settle on introduced a deflated set of manifests that use numeric prefixes to convey ordering. We can iterate going forward as we expand the controller logic anyways. In case anyone is interested, this is a real simple, concrete example of what I had suggested: https://github.com/timflannagan/k8s-scripts/blob/master/scripts/create-nfs-provisioner.sh#L25-L27.

Reply:
Okay cool, I've got this setup. Let me know what you think in the next push.